jetty-util/src/test/java/org/eclipse/jetty/util/thread/QueuedThreadPoolTest.java

```diff
@@ -505,6 +504,57 @@ public void testShrink() throws Exception
         tp.stop();
     }
 
+    @Test
+    public void testSteadyShrink() throws Exception
+    {
+        CountDownLatch latch = new CountDownLatch(1);
+        Runnable job = () ->
+        {
+            try
+            {
+                latch.await();
+            }
+            catch(InterruptedException e)
+            {
+                e.printStackTrace();
+            }
+        };
+
+        QueuedThreadPool tp = new QueuedThreadPool();
+        tp.setMinThreads(2);
+        tp.setMaxThreads(10);
+        tp.setIdleTimeout(500);
+        tp.setThreadsPriority(Thread.NORM_PRIORITY - 1);
+
+        tp.start();
+        waitForIdle(tp, 2);
+        waitForThreads(tp, 2);
+
+        for (int i = 0; i < 10; i++)
+            tp.execute(job);
+
+        waitForThreads(tp, 10);
+        int threads = tp.getThreads();
+        // let the jobs run
+        latch.countDown();
+
+        for (int i = 5; i-- > 0; )
+        {
+            Thread.sleep(250);
```

Comment:
This should be `timeout / 2` to clearly link the fact that you are sleeping a time that is long enough to idle some thread, but not all of them.